### ISSUE TITLE
Ice Breaker

### ISSUE THREAD

**Issue by lite20** [NodeOS/NodeOS#211]
So there has been little discussion about this project, but I know a few of you have heard of it and maybe got hints about it from say https://github.com/NodeOS/NodeOS/issues/178 and I've decided I'd finally shed some light on the Ice Breaker project.

So Ice Breaker, is my experimental challenge I began about a year ago. The Ice Breaker itself, is an open-hardware mobile device. The goal of the project has been to, is to take the power and freedom we "the user" have had with computers, and bring it to the mobile scene. The phone is designed to be hackable and features a modular "plug and play" motherboard I've been designing for the last few months.

So where does NodeOS come into this? As you probably guessed by now, NodeOS is my current target platform for the device. Of course, I'll probably create an Android setup too for users who just want a good phone, but the NodeOS platform places in the hands of every developer who know's JS the ability to alter every aspect of the device. The ability to make those "I wish I could" dreams of their phone come true. With npm, modules can simply be downloaded and extending functionality of the phone in, by example, swapping out the camera for a higher resolution, would be as simple as swapping the hardware, and downloading a new module to handle it.

I decided to share Ice Breaker in case it come as a cool inspiration to anyone. At the moment, I'm the only one chipping away slowly at the project, but am aiming, as soon as I can, to create a web page for it and let the community get involved in it's progress.

**Comment by formula1** [comment]
https://github.com/Silk-GUI/Silk/ @zodern will likely be interested. Are you making a dev board? Are you using a rasberry pi?

**Comment by Coretool** [comment]
@lite20 I like your idea :) I had the same some months ago (and did some research, etc.) Is there a way to help you ^^ I would **love** to help ! BTW your names are epic ;)

**Comment by piranna** [comment]
Take in account that currently NodeOS is x86 only, I've been able to
compile it for ARM but was not able to make the kernel to boot a simple
init program...

**Comment by Coretool** [comment]
@piranna Problems with the bootloader or what ?

**Comment by lite20** [comment]
@formula1 yes for the prototypes I'm using modified raspberries and slowly it will evolve. With the raspberry Pi zero, I'll be tending to work more with it as with the modular mother board, its less painful if I break it. I do eventually plan to evolve to a more powerful board though

**Comment by lite20** [comment]
@formula and there will be fairly descriptive instructions on how to mod the pi to make a prototype to play with fyi :) (just updating them to work with the Pi zero)

**Comment by piranna** [comment]
> @piranna Problems with the bootloader or what ?

Kernel boots, but when executing the init it throw a kernel panic with a
"init not tainted" error... :-/

**Comment by Coretool** [comment]
@piranna I asume that there is no selinux, right ? Because if selinux is there, it may cause this error... However, this error has many reasons to occur

**Comment by piranna** [comment]
> @piranna I asume that there is no selinux, right ? Because if selinux is
there, it may cause this error... However, this error has many reasons to
occur

I'm not using SELinux, but looning for other things I've found a trace on
the boot log of x86 that says "SELinux: Registering netfilter hooks". I
didn't know it was being enabled by default in vanilla kernels, do you
suggest to explicitly disable it alltogether?

**Comment by Coretool** [comment]
AFAIK is this error often caused by Selinux releated stuff... 
Try to disable everything that is related to Selinux and try it again .  (I think it is a kernel argumen.) elinux is on by default (I think) and causes / caused many problems this way on RHEL and FedoraSince I don't know your complete config, I can't help you more ...

**Comment by lite20** [comment]
> @lite20 I like your idea :) I had the same some months ago (and did some research, etc.) Is there a way to help you ^^ I would love to help ! BTW your names are epic ;)

**Comment by lite20** [comment]
@Coretool awesome! Thanks on the naming haha I come from a background of game development. I'd more than love to have you contribute! I'm working on a website to organize the efforts so that we can progress smoothly toward our goal.

**Comment by lite20** [comment]
> Take in account that currently NodeOS is x86 only, I've been able to
compile it for ARM but was not able to make the kernel to boot a simple
init program...

As you always say, one step at a time :) With an ARM SoC I'll be able to help you debug, and improve NodeOS for that platform and maybe also contribute to making it run smoother on memory deprived platforms.
